src/main/kotlin/inc/evil/d17map/security/user/UserAuthService.kt

```diff
@@ -0,0 +1,51 @@
+package inc.evil.d17map.security.user
+
+import inc.evil.d17map.entities.User
+import inc.evil.d17map.enums.Role
+import inc.evil.d17map.repositories.UserRepository
+import inc.evil.d17map.security.jwt.JWTService
+
+
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.Authentication
+import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.stereotype.Service
+
+@Service
+class UserAuthService(
+    private val userRepository: UserRepository,
+    private val passwordEncoder: PasswordEncoder,
+    private val authenticationManager: AuthenticationManager,
+    private val jwtService: JWTService
+) {
+
+    // TODO user reservations during registration seem to be awkward :/
+    fun registerUser(registerRequest: AuthRequest): User {
+        val user = User(
+            email = registerRequest.username,
+            password = passwordEncoder.encode(registerRequest.password),
+            reservations = mutableSetOf(),
+            userType = Role.STUDENT,
+        )
+        return userRepository.save(user)
+    }
```

Comment:
Przydałoby się sprawdzać, czy przypadkiem dany user (z danym adresem e-mail) nie ma już założonego konta.

``````````
fun registerUser(registerRequest: AuthRequest): User {
    if (userRepository.existsByEmail(registerRequest.username)) {
        throw RuntimeException("User already exists with email: ${registerRequest.username}")
    }

    val user = User(
        email = registerRequest.username,
        password = passwordEncoder.encode(registerRequest.password),
        reservations = mutableSetOf(),
        userType = Role.STUDENT,
    )
    return userRepository.save(user)
}

Reply:
Zrobione, ale trzeba dodać w nowym tasku `UserAlreadyExistsException`.

Reply:
Ok, zrobimy to w kolejnym Tasku, przy okazji z innymi exceptions